great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -1608,6 +1660,70 @@ def add_store(self, store_name: str, store_config: StoreConfigTypedDict) -> Stor
         self._save_project_config()
         return store
 
+    @public_api
+    @new_method_or_class(version="0.17.2")
+    def add_data_docs_site(
+        self, site_name: str, site_config: DataDocsSiteConfigTypedDict
+    ) -> None:
+        """Add a new Data Docs Site to the DataContext.
+
+        Example site config dicts can be found in our "Host and share Data Docs" guides.
+
+        Args:
+            site_name: New site name to add.
+            site_config: Config dict for the new site.
+        """
+        if site_name in self.config.data_docs_sites:
+            raise ValueError(
+                f"Data Docs Site `{site_name}` already exists in the Data Context."
+            )
+
+        sites = self.config.data_docs_sites
+        sites[site_name] = site_config
+        self.variables.data_docs_sites = sites
+        self._save_project_config()
+
+    @public_api
+    @new_method_or_class(version="0.17.2")
+    def update_data_docs_site(
+        self, site_name: str, site_config: DataDocsSiteConfigTypedDict
+    ) -> None:
+        """Update an existing Data Docs Site.
+
+        Example site config dicts can be found in our "Host and share Data Docs" guides.
+
+        Args:
+            site_name: Site name to update.
+            site_config: Config dict that replaces the existing.
+        """
+        if site_name not in self.config.data_docs_sites:
+            raise ValueError(
+                f"Data Docs Site `{site_name}` does not already exist in the Data Context."
+            )
+
+        sites = self.config.data_docs_sites
+        sites[site_name] = site_config
+        self.variables.data_docs_sites = sites
+        self._save_project_config()
+
+    @public_api
+    @new_method_or_class(version="0.17.2")
+    def delete_data_docs_site(self, site_name: str):
+        """Delete an existing Data Docs Site.
+
+        Args:
+            site_name: Site name to delete.
+        """
+        if site_name not in self.config.data_docs_sites:
```

Comment:
This might be idempotent - check consistency with datasources.

Reply:
Currently the data context methods do raise an exception if the item to be deleted does not exist. Here is the behavior of each (listed in order they appear in abstract_data_context.py):
- NEW: `delete_data_docs_site()`: Raise ValueError in body in body of method if key (name) doesn't exist (will change error type to InvalidKeyError).
- `delete_store()`: Raise StoreConfigurationError in body in body of method if key (name) doesn't exist.
- `delete_datasource()`: Raise ValueError in datasource_store on `get_datasource()` call if key (name) doesn't exist.
- `delete_checkpoint()`: Raise InvalidKeyError in checkpoint_store if key (name, id) doesn't exist.
- `delete_expectation_suite()`: DataContextError raised in body of method if key (name) doesn't exist.
- `delete_profiler()`: InvalidKeyError raised in `delete_profiler` in rule_based_profiler.py if key (name or id) doesn't exist

cc @billdirks since we had discussed.